activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb

```diff
@@ -172,6 +177,21 @@ def table_comment(table_name) # :nodoc:
           end
         end
 
+        # Returns the partition definition of a given table
+        def table_partition_definition(table_name) # :nodoc:
+          scope = quoted_scope(table_name, type: "BASE TABLE")
+          return unless scope[:name]
```

Comment:
Can the `scope[:name]` actually be `nil` here?

Reply:
i'm not sure - i was just preserving the pattern above of only performing this query based on the return of `quoted_scope`, as happens above in `table_comment`

Reply:
You can just remove it.